### PR TITLE
FUSETOOLS2-2200: Testing with productized versions of Camel

### PR DESCRIPTION
**Pull request by pospisilf**

### CODE REVIEW

File: src/ui-test/utils.ts [hunk 1]
```diff
@@ -484,3 +491,92 @@ export async function selectTask(driver: WebDriver, task: string): Promise<void>
...
+    }
+}
+
+/**
+ * Compare two versions in format "^\d+(\.\d+)*$".
+ * @param base Base version.
+ * @param target Version to be compared with base version.
+ * @returns true if target is newer or same as base, false otherwise
+ */
+export function compareVersions(base: string, target: string): boolean {
```
Comment (apupier) [inline src/ui-test/utils.ts:570]: I think we can find a more precise naming, somethign like `isVersionNewer` or `isVersionHigher`

Comment (pospisilf) [reply src/ui-test/utils.ts]: You're right. Changed.

**Review by apupier — CHANGES_REQUESTED** [review]
tab indentation should be used, it is mor einclusive as it permits the users to choose the spacing that they want on their display screen and screen readers are better with it. (also better at semantic level)

**Comment by pospisilf** [comment]
> tab indentation should be used, it is mor einclusive as it permits the users to choose the spacing that they want on their display screen and screen readers are better with it. (also better at semantic level)

Can you be more specific, I probably don't understand what you mean by this in this PR. @apupier

**Comment by djelinek** [comment]
> > tab indentation should be used, it is mor einclusive as it permits the users to choose the spacing that they want on their display screen and screen readers are better with it. (also better at semantic level)
> 
> Can you be more specific, I probably don't understand what you mean by this in this PR. @apupier

my guess are these and few following changes - https://github.com/camel-tooling/camel-dap-client-vscode/pull/581/files#diff-0d36ac6cda0dcf10be862d40f72a4ff5517868653e4da383b013046a69c8ceceR179

it was done probably automatically because of some local workspace or user settings you are using.. there is no .editorconfig which would define spacing for this project so that can happen..

File: src/ui-test/tests/codelens.test.ts [hunk 2]
```diff
@@ -73,7 +74,11 @@ describe('JBang commands execution through command codelens', function () {
     });
 
     it(`Execute command 'apache.camel.debug.jbang' with codelens '${variables.CAMEL_DEBUG_CODELENS}'`, async function () {
-        await (await findCodelens(variables.CAMEL_DEBUG_CODELENS)).click();
+        if (process.env.CAMEL_VERSION !== undefined && isCamelVersionProductized(process.env.CAMEL_VERSION)){
```
Comment (djelinek) [inline src/ui-test/tests/codelens.test.ts:77]: this statement can be simplified if you move `process.env.CAMEL_VERSION !== undefined ` directly into `isCamelVersionProductized`

**Review by djelinek — CHANGES_REQUESTED** [review]

File: src/ui-test/env/set.camel.version.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+		// no env variable set or is empty
+		if (process.env.CAMEL_VERSION === undefined || process.env.CAMEL_VERSION === null || process.env.CAMEL_VERSION.length === 0) {
+			this.skip();
+		}
+
+		// set version directly
+		setUserSettingsDirectly(CATALOG_VERSION_ID, process.env.CAMEL_VERSION);
+
+		// match jbang
+		setUserSettingsDirectly(JBANG_VERSION_ID, extractVersionNumber(process.env.CAMEL_VERSION));
```
[the remark this answers is not in the document]
Comment (apupier) [reply src/ui-test/env/set.camel.version.ts]: ideally yes.
Maybe in some case we will require the match

**Comment by djelinek** [comment]
Depends on #594

**Comment by djelinek** [comment]
@pospisilf please do a rebase on main branch, changes:

- bump jbang from 4.3.0 to 4.4.0
- fix for attaching debugger with used 4.0.0 camel version

File: src/ui-test/uitest_runner.ts [hunk 4]
```diff
@@ -28,7 +28,12 @@ const extensionFolder = variables.EXTENSION_DIR;
 
 async function main(): Promise<void> {
 	const tester = new ExTester(storageFolder, releaseType, extensionFolder);
-	await tester.setupAndRunTests('out/ui-test/tests/**/*.test.js',
+	await tester.setupAndRunTests(
+		[
+			'out/ui-test/env/set.camel.version.js',
+			'out/ui-test/tests/codelens.test.js',
```
Comment (djelinek) [inline src/ui-test/uitest_runner.ts:34]: should use `*.test.js`

**Review by djelinek — CHANGES_REQUESTED** [review]

**Review by djelinek — CHANGES_REQUESTED** [review]
please double check why a lot of tests are skipped also on main pipeline, see eg https://github.com/camel-tooling/camel-dap-client-vscode/actions/runs/7979338290/job/21786524988?pr=581#step:14:81

File: .github/workflows/camel_version.yaml [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+        if: matrix.os == 'windows-latest'
+        run: choco install jbang
+
+      - name: Setup JBang (trusted sources)
+        run: jbang trust add https://github.com/apache/
+
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: "18.15.x"
```
Comment (djelinek) [inline .github/workflows/camel_version.yaml:59]: from now on we can use just '18.x'

**Review by djelinek — CHANGES_REQUESTED** [review]

**Pull request merged**